### ISSUE TITLE
Uncaught ReferenceError: body is not defined

### ISSUE THREAD

**Issue by stianl** [mcstudios/glightbox#49]
The removal of the body variable in b9daaf329f05cdc3e334251e034e814be4951d0b might be causing some issues with touch navigation? I get this error when I attempt to swipe in Chrome device emulation:

```javascript
glightbox.js:1180 Uncaught ReferenceError: body is not defined
    at withCallback (glightbox.js:1180)
    at HTMLDocument.handler (glightbox.js:327)
```

Setting var `body = document.body` in the `touchNavigation()` function solves the problem.

**Comment by TeachMeHowToUse2FA** [comment]
Navigation on a mobile device does not work at all. Even on demo site. Is there any way we can solve this problem?

**Comment by fnicastri** [comment]
any update on this issue?

**Comment by designast** [comment]
> any update on this issue?

1. Download unminify glightbox.js
2. Open and find function `function touchNavigation()`
3. Add inside this function code `var body = document.body;`
4. After last Chrome update need to fix scroll and touchmove event which use `event.preventdefault(); `
4.1 In function  `function touchNavigation()` find:
```
        this.events['doctouchmove'] = addEvent('touchmove', {
            onElement: document,
            withCallback: function withCallback(e, target) {
                if (hasClass(body, 'gdesc-open')) {
                    e.preventDefault();
                    return false;
                }
            }
        });
```
And change event to this(we add the last parameter - `useCapture: { passive: false }` ):
```
        this.events['doctouchmove'] = addEvent('touchmove', {
            onElement: document,
            withCallback: function withCallback(e, target) {
                if (hasClass(body, 'gdesc-open')) {
                    e.preventDefault();
                    return false;
                }
            },
            useCapture: { passive: false }
        });
```
4.2 In function  `function touchNavigation()` find:
`this.events['touchMove'] = addEvent('touchmove', { ... ... ...`
And change event to this:
 ```
       this.events['touchMove'] = addEvent('touchmove', {
            onElement: body,
            withCallback: function withCallback(e, target) {
                if (!hasClass(body, 'touching')) {
                    return;
                }
                if (hasClass(body, 'gdesc-open') || doingPinch || pinchBigger) {
                    return;
                }
                e.preventDefault();
                endCoords = e.targetTouches[0];
                var slideHeight = activeSlide.querySelector('.gslide-inner-content').offsetHeight;
                var slideWidth = activeSlide.querySelector('.gslide-inner-content').offsetWidth;

                var xUp = e.targetTouches[0].clientX;
                var yUp = e.targetTouches[0].clientY;
                var xDiff = xDown - xUp;
                var yDiff = yDown - yUp;

                if (Math.abs(xDiff) > Math.abs(yDiff)) {
                    /*most significant*/
                    vSwipe = false;
                    hSwipe = true;
                } else {
                    hSwipe = false;
                    vSwipe = true;
                }

                if (vSwipe) {
                    vDistanceLast = vDistance;
                    vDistance = endCoords.pageY - startCoords.pageY;
                    if (Math.abs(vDistance) >= vSwipMinDistance || vSwipe) {
                        var opacity = 0.75 - Math.abs(vDistance) / slideHeight;
                        activeSlideMedia.style.opacity = opacity;
                        if (activeSlideDesc) {
                            activeSlideDesc.style.opacity = opacity;
                        }
                        slideCSSTransform(activeSlideMedia, 'translate3d(0, ' + vDistance + 'px, 0)');
                    }
                    return;
                }

                hDistanceLast = hDistance;
                hDistance = endCoords.pageX - startCoords.pageX;
                hDistancePercent = hDistance * 100 / winWidth;

                if (hSwipe) {
                    if (_this4.index + 1 == _this4.elements.length && hDistance < -60) {
                        resetSlideMove(activeSlide);
                        return false;
                    }
                    if (_this4.index - 1 < 0 && hDistance > 60) {
                        resetSlideMove(activeSlide);
                        return false;
                    }

                    var _opacity = 0.75 - Math.abs(hDistance) / slideWidth;
                    activeSlideMedia.style.opacity = _opacity;
                    if (activeSlideDesc) {
                        activeSlideDesc.style.opacity = _opacity;
                    }
                    slideCSSTransform(activeSlideMedia, 'translate3d(' + hDistancePercent + '%, 0, 0)');
                }
            },
            useCapture: { passive: false }
        });

```

5. Minify js file if you need.

**Comment by mcstudios** [comment]
This issue has been fixed in the latest version 2.0.1, please download again and try it.

**Comment by stianl** [comment]
Works fine with 2.0.1! :+1:
